internal/service/base/resource_language_translation.go

```diff
@@ -0,0 +1,573 @@
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	// Read response into the model
+	resp.Diagnostics.Append(data.readClientResponse(*responseData, data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *languageTranslationResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data languageTranslationsResourceModel
+
+	if r.Client == nil || r.Client.ManagementAPIClient == nil {
+		resp.Diagnostics.AddError(
+			"Client not initialized",
+			"Expected the PingOne client, got nil.  Please report this issue to the provider maintainers.")
+		return
+	}
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update API call logic
+	clientData, diags := data.buildClientStruct()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// The UpdateTranslations API call does not return a response
+	resp.Diagnostics.Append(framework.ParseResponse(
+		ctx,
+
+		func() (any, *http.Response, error) {
+			fO, fR, fErr := r.Client.ManagementAPIClient.TranslationsApi.UpdateTranslations(ctx, data.EnvironmentId.ValueString(), data.Locale.ValueString()).LocaleTranslation(*clientData).Execute()
+			return framework.CheckEnvironmentExistsOnPermissionsError(ctx, r.Client.ManagementAPIClient, data.EnvironmentId.ValueString(), fO, fR, fErr)
+		},
+		"UpdateTranslations-Update",
+		framework.DefaultCustomError,
+		sdk.DefaultCreateReadRetryable,
+		nil,
+	)...)
+
+	// Subsequent read needed as the API does not return the full object on any create or update operation.
+	var responseData *[]management.LocaleTranslation
+	resp.Diagnostics.Append(framework.ParseResponse(
+		ctx,
+
+		func() (any, *http.Response, error) {
+			pagedIterator := r.Client.ManagementAPIClient.TranslationsApi.ReadTranslations(ctx, data.EnvironmentId.ValueString(), data.Locale.ValueString()).Execute()
+			var initialHttpResponse *http.Response
+
+			translationsSlice := make([]management.LocaleTranslation, 0)
+			for pageCursor, err := range pagedIterator {
+				if err != nil {
+					return framework.CheckEnvironmentExistsOnPermissionsError(ctx, r.Client.ManagementAPIClient, data.EnvironmentId.ValueString(), nil, pageCursor.HTTPResponse, err)
+				}
+
+				if initialHttpResponse == nil {
+					initialHttpResponse = pageCursor.HTTPResponse
+				}
+
+				if translations, ok := pageCursor.EntityArray.Embedded.GetTranslationsOk(); ok {
+					// Append the translations to the slice
+					translationsSlice = append(translationsSlice, translations...)
+				} else {
+					return nil, pageCursor.HTTPResponse, fmt.Errorf("no translations found in response")
+				}
+			}
+
+			return &translationsSlice, initialHttpResponse, nil
+		},
+		"UpdateTranslations-Create-SubsequentRead",
```

Comment:
Should this be UpdateTranslations-Update? Currently it matches the Create method